eval/src/vespa/eval/tensor/sparse/sparse_tensor_index.cpp

```diff
@@ -199,10 +199,24 @@ SparseTensorIndex::SparseTensorIndex(size_t num_mapped_in)
     : _stash(STASH_CHUNK_SIZE), _map(), _num_mapped_dims(num_mapped_in)
 {}
 
-SparseTensorIndex::SparseTensorIndex(const SparseTensorIndex & index_in)
-    : _stash(needed_memory_for(index_in)), _map(), _num_mapped_dims(index_in._num_mapped_dims)
+SparseTensorIndex::SparseTensorIndex(size_t stash_size, const SparseTensorIndex &index_in)
+    : _stash(stash_size), _map(), _num_mapped_dims(index_in._num_mapped_dims)
 {
-    copyMap(_map, index_in._map, _stash);
+    copyMap(_map, index_in._map, _stash);    
+}
+
+SparseTensorIndex
+SparseTensorIndex::shrunk_copy() const
+{
+    size_t want_mem = needed_memory_for(*this);
```

Comment:
consider renaming the function calculating the wanted chunk size to make us feel more safe about not always wanting to put everything into a single chunk.